shell/platform/android/io/flutter/embedding/engine/android/FlutterView.java

```diff
@@ -79,16 +91,78 @@ private void init() {
       case surface:
         Log.d(TAG, "Internally creating a FlutterSurfaceView.");
         FlutterSurfaceView flutterSurfaceView = new FlutterSurfaceView(getContext());
+        renderSurface = flutterSurfaceView;
         addView(flutterSurfaceView);
         break;
       case texture:
         Log.d(TAG, "Internally creating a FlutterTextureView.");
         FlutterTextureView flutterTextureView = new FlutterTextureView(getContext());
+        renderSurface = flutterTextureView;
         addView(flutterTextureView);
         break;
     }
   }
 
+  /**
+   * Connects this {@code FlutterView} to the given {@link FlutterEngine}.
+   *
+   * Once invoked, the Flutter UI painted by the given {@link FlutterEngine} will be
+   * displayed by this {@code FlutterView}. Additionally, user touch events, accessibility
+   * events, keyboard events, and more will be forwarded from this {@code FlutterView}
+   * to the attached {@link FlutterEngine}.
+   *
+   * See {@link #detachFromFlutterEngine()} for information on how to detach from a
+   * {@link FlutterEngine}.
+   */
+  public void attachToFlutterEngine(@NonNull FlutterEngine flutterEngine) {
+    if (isAttachedToFlutterEngine()) {
+      if (flutterEngine == this.flutterEngine) {
+        // We are already attached to this FlutterEngine
+        return;
+      }
+
+      // Detach from a previous FlutterEngine so we can attach to this new one.
+      detachFromFlutterEngine();
+    }
+
+    this.flutterEngine = flutterEngine;
+
+    // Instruct our FlutterRenderer that we are now its designated RenderSurface.
+    this.flutterEngine.getRenderer().attachToRenderSurface(renderSurface);
+  }
+
+  /**
+   * Disconnects this {@code FlutterView} from a previously attached {@link FlutterEngine}.
+   *
+   * Once invoked, the UI of this {@code FlutterView} will be cleared, and all touch events,
+   * accessibility events, keyboard events, etc. will no longer be forwarded to the previously
```

Comment:
> will no longer be forwarded to the previously attached FlutterEngine

Suggestion: this sentence would be clearer without the passive voice.  Who is forwarding events?  Is it this FlutterView?  If so, say so:

> This {FlutterView} will stop forwarding all events to the previously-attached {FlutterEngine}.  This includes touch events, accessibility events, keyboard events, and others.

In general, I think it is harder to write with active voice when writing documentation.  However, the active voice can be more readable and more concise than the passive voice.

Reply:
Done